hawkular-tenant-jaxrs-filter/src/main/java/org/hawkular/jaxrs/filter/tenant/TenantFeature.java

```diff
@@ -24,10 +24,16 @@
 import javax.ws.rs.ext.Provider;
 
 /**
+ * A JAX-RS dynamic feature that adds the {@link TenantFilter} to all methods which have {@link TenantRequired}
+ * effectively as "true", which might be the default value for {@link TenantRequired}. In the usual case, this feature
+ * is added to all classes and methods, except when classes/methods are annotated with "false".
+ *
  * @author Juraci Paixão Kröhling
  */
 @Provider
 public class TenantFeature implements DynamicFeature {
+    private static final TenantFilter TENANT_FILTER = new TenantFilter();
```

Comment:
Yes. Constants are resolved at classloading time (basically in a synthetic static initializer), which is guaranteed to be called exactly once by the JVM.